File: Sorting and Searching/concert_tickets.py
"""
    Author: JN.Wang
    Created: 03/10/24
"""

from sys import stdin, stdout
input = stdin.readline
# print = stdout.write

'''
    https://cses.fi/problemset/task/1091
'''

from bisect import bisect_right

def main():
    n, m = map(int, input().split())
    
    ticket_prices = sorted([int(x) for x in input().split()])
    max_prices = [int(x) for x in input().split()]
    
    used = [False for _ in range(m + 2)]
    for _ in range(m):
        ticket = bisect_right(ticket_prices, max_prices[_]) - 1
        
        while ticket > 0 and used[ticket] is True:
            ticket -= 1
        
        if ticket <= -1:
            print(ticket)
        elif ticket_prices[ticket] <= max_prices[_] and used[ticket] is False:
            print(ticket_prices[ticket])
            used[ticket] = True
        else:
            print(-1)
        
if __name__ == "__main__":
    main()
    

File: Sorting and Searching/concert-tickets.cpp
/*
    Author: soju-jsh
    Created: 03.28.24

    Problem: https://cses.fi/problemset/task/1091

    Result: AC
*/

#include <bits/stdc++.h>
using namespace std;

// Datatype Shortcuts
using ll = long long;
using vll = vector<ll>;
using vvll = vector<vll>;
using pll = pair<ll, ll>;
 
// Chars
#define nl '\n'
#define wp ' '
 
// Macros
#define ALL(x) x.begin(), x.end()
#define rALL(x) x.rbegin(), x.rend()
#define REP(i, ii, t) for (ll i = ii; i < t; i++)
#define rREP(i, t) for (ll i = t - 1; i >= 0; i--)
#define vll_cin(arr) for (auto &i : arr) cin >> i;
#define test_case() ll tt; cin >> tt; while (tt--)

// Debugging purposes
#define display(arr) for (auto i = arr.begin(); i != arr.end(); i++) cout << *i << wp; cout << nl;

void fast(){
    ios::sync_with_stdio(false);
    cin.tie(NULL);
}

int main(){
    fast();

    // freopen("input.txt", "r", stdin);
    // freopen("output.txt", "w", stdout);
    
    ll n, m; 
    cin >> n >> m;

    multiset<ll> tickets;
    REP(i, 0, n) {
        ll price; cin >> price;
        tickets.emplace(price);
    }

    REP(i, 0, m) {
        ll willing_price; cin >> willing_price;

        auto match = tickets.lower_bound(willing_price);
        if (tickets.empty() or (*match > willing_price and match == tickets.begin())) cout << -1 << nl;
        else
        {
            if (*match != willing_price) match--;
            cout << *match << nl;
            tickets.erase(match);
        }
    }
}